src/mqtt.c

```diff
@@ -1084,13 +1084,11 @@ ssize_t mqtt_pack_connection_request(uint8_t* buf, size_t bufsz,
     remaining_length = 10; /* size of variable header */
 
     if (client_id == NULL) {
-        /* client_id is a mandatory parameter */
-        return MQTT_ERROR_CONNECT_NULL_CLIENT_ID;
-    } else {
-        /* mqtt_string length is strlen + 2 */
-        remaining_length += __mqtt_packed_cstrlen(client_id);
+        client_id = "";
```

Comment:
> If the Client supplies a zero-byte ClientId, the Client MUST also set CleanSession to 1 [MQTT-3.1.3-7].

There should be an error check here that `MQTT_CONNECT_CLEAN_SESSION` is set. Maybe a new error like "`MQTT_ERROR_CLEAN_SESSION_IS_REQUIRED`".

Reply:
Good point.

Reply:
Would it be acceptable to re-use the now deprecated member, or do you prefer a new entry?